aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,140 @@
-.. include:: /../common/stub-topic.txt
+Routing to Controller Actions
+=============================
 
-|stub-icon| Routing to Controller Actions
-=========================================
+By `Rachel Appel`_
 
-.. include:: /../common/stub-notice.txt
+In this article:
 
-.. _issue: https://github.com/aspnet/Docs/issues/117
+.. contents:: Sections
+  :local:
+  :depth: 1
+
+Relationship to middleware
+--------------------------
+
+In the early days of the web you didn’t think about routing, as most web sites or applications simply referenced folders and files on a disk. If you had a URL for ``http://contoso.com/movies/GoneWithTheWind.htm``, then there was a folder called ``movies`` with a file called ``GoneWithTheWind.htm`` on the disk. With modern web frameworks like MVC you do not have to structure our content or application exactly how the world will see it. You can present it however you choose, and organize your software in the way that makes sense. `Routing <https://docs.asp.net/en/latest/fundamentals/routing.html>`_ is what you use to connect our code to the URL and features our application exposes. 
+
+Routing is part of ASP.NET Core middleware. Middleware is a component that is assembled into an application pipeline to handle requests and responses. It's the middleware that processes the request until it gets to routing, which then hands over the request to MVC. MVC then selects an action method based on data that comes from routing. After that point it's up to the developer to add code to the action method to retrieve and modify data, or call services. 
```

Comment:
What's the document is already pretty accurate accurate.

> MVC then selects an action method based on data that comes from routing.

This is mostly correct, but often other criteria are used in addition to the route data (such as HTTP verb).